File: WebApp/WebApp/Models/PaymentDetails.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace WebApp.Models
{
    public class PaymentDetails
    {
        [Key]
        public string Id { get; set; }
        [Required]
        public string Cart { get; set; }
        public string CreateTime { get; set; }
        public string Intent { get; set; }
        [Required]
        public string State { get; set; }
        public string PayerName { get; set; }
        public string PayerSurname { get; set; }
        [Required]
        public string PayerId { get; set; }
        [EmailAddress]
        public string PayerEmail { get; set; }
        [Required]
        public string AddressCountryCode { get; set; }
        public string AddressCity { get; set; }
        public string AddressLine { get; set; }
        [Required]
        public string AddressPostalCode { get; set; }
        public string ItemName { get; set; }
        [Required]
        public Double ItemPrice { get; set; }
        [Required]
        public string ItemCurrency { get; set; }
        public Double CurrencyRate { get; set; }
        //public Guid SoldTicketId { get; set; }
        //public SoldTicket SoldTicket { get; set; }
    }
}

File: angular-app/src/models/payment-details.ts
export class PaymentDetails{
    public Id: string; 
    public Cart: string; 
    public CreateTime: string; 
    public Intent: string; 
    public State: string; 
    public PayerName: string; 
    public PayerSurname: string; 
    public PayerId: string; 
    public PayerEmail: string; 
    public AddressCountryCode: string; 
    public AddressCity: string; 
    public AddressLine:string; 
    public AddressPostalCode: string; 
    public ItemName: string; 
    public ItemPrice: number; 
    public ItemCurrency: string; 
    public CurrencyRate: number;

    constructor(data: any){
        this.Id = data['id'];
        this.Cart = data['cart'];
        this.CreateTime = data['create_time'];
        this.Intent = data['intent'];
        this.State = data['state'];
        this.PayerName = data['payer']['payer_info']['first_name'];
        this.PayerSurname = data['payer']['payer_info']['last_name'];
        this.PayerId = data['payer']['payer_info']['payer_id'];
        this.PayerEmail = data['payer']['payer_info']['email'];
        this.AddressCountryCode = data['payer']['payer_info']['shipping_address']['country_code'];
        this.AddressCity = data['payer']['payer_info']['shipping_address']['city'];
        this.AddressLine = data['payer']['payer_info']['shipping_address']['line1'];
        this.AddressPostalCode = data['payer']['payer_info']['shipping_address']['postal_code'];
        this.ItemName = data['transactions']['0']['item_list']['items']['0']['name'];
        this.ItemPrice = data['transactions']['0']['item_list']['items']['0']['price'];
        this.ItemCurrency = data['transactions']['0']['item_list']['items']['0']['currency'];
    }
}